sdmetrics/reports/utils.py

```diff
@@ -479,6 +479,133 @@ def get_column_pair_plot(real_data, synthetic_data, column_names, metadata):
+
+    Args:
+        data (pandas.DataFrame):
+            The cardinality data.
+        parent_primary_key (string):
+            The name of the primary key column in the parent table.
+        child_foreign_key (string):
+            The name of the foreign key column in the child table.
+
+    Returns:
+        plotly.graph_objects._figure.Figure
+    """
+    fig = px.histogram(data,
+                       x='# children',
+                       y='# parents',
+                       color='data',
+                       barmode='group',
+                       color_discrete_sequence=[DATACEBO_DARK, DATACEBO_LIGHT],
+                       pattern_shape='data',
+                       pattern_shape_sequence=['', '/'],
+                       nbins=max(data['# children']) - min(data['# children']) + 1,
+                       histnorm='probability density')
+
+    for name in ['Real', 'Synthetic']:
+        fig.update_traces(
+            hovertemplate=f'<b>{name}</b><br>Frequency: %{{y}}<extra></extra>',
+            selector={'name': name}
+        )
+
+    title = (
+        f"Relationship (child foreign key='{child_foreign_key}' and parent "
+        f"primary key='{parent_primary_key}')"
+    )
+    fig.update_layout(
+        title=title,
+        xaxis_title='# of Children (per Parent)',
+        yaxis_title='Frequency',
+        plot_bgcolor=BACKGROUND_COLOR
+    )
+    return fig
+
+
+def get_cardinality_plot(real_data, synthetic_data, child_table_name, parent_table_name,
+                         child_foreign_key, metadata):
+    """Return a plot of the cardinality of the parent-child relationship.
+
+    Args:
+        real_data (dict):
+            The real data.
+        synthetic_data (dict):
+            The synthetic data.
+        child_table_name (string):
+            The name of the child table.
+        parent_table_name (string):
+            The name of the parent table.
+        child_foreign_key (string):
+            The name of the foreign key column in the child table.
+        metadata (dict):
+            The metadata.
+
+    Returns:
+        plotly.graph_objects._figure.Figure
+    """
+    parent_primary_key = None
+    for relation_dict in metadata.get('relationships', []):
+        parent_match = relation_dict['parent_table_name'] == parent_table_name
+        child_match = relation_dict['child_table_name'] == child_table_name
+        foreign_key_match = relation_dict['child_foreign_key'] == child_foreign_key
+        if child_match and parent_match and foreign_key_match:
+            parent_primary_key = relation_dict['parent_primary_key']
+
+    if parent_primary_key is None:
+        raise ValueError(
+            f"Relationship between child table '{child_table_name}' and parent table "
+            f"'{parent_table_name}' for the foreign key '{child_foreign_key}' "
+            'not found in the metadata. Please update the metadata.'
```

Comment:
minor: can we reword this to say "No relationship found between... in the metadata. Please update the metadata"

Reply:
Yes, done in [76fcf86](https://github.com/sdv-dev/SDMetrics/pull/384/commits/76fcf862286df85acac9ee2977216ac5c72f5a30)